gammapy/makers/map.py

```diff
@@ -75,6 +78,35 @@ def make_counts(geom, observation):
         counts = Map.from_geom(geom)
         counts.fill_events(observation.events)
         return counts
+    
+    @staticmethod
+    def make_counts_rad_max(geom, observation):
```

Comment:
I agree the method make sense in terms of code organisation, but maybe not that much in terms of API? I think ideally the handling of energy dependent region sizes could be almost hidden completely to the user. E.g. like:

```python3
from gammapy.makers.utils import make_counts_rad_max

def make_counts(geom, observation):
      
      # TODO: Maybe add RegionGeom.is_energy_dependent or similar....
      if geom.is_region and isinstance(geom.region, PointSkyRegion):
          counts = make_counts_rad_max(geom, rad_max, events)
      else:
           counts.fill_events()

```